service/controller/resource/subnet/resource.go

```diff
@@ -233,6 +236,16 @@ func (r *Resource) ensureSubnets(ctx context.Context, deploymentsClient *azurere
 		}
 	}
 
+	// Update AzureCluster so that subnet.ID is saved.
```

Comment:
We save the `AzureCluster` object back to the k8s API after finish iterating all the subnets. In case we hit a conflict we log the error but don't return it. On the next iteration the handler will try to save it again.